File: dependencies/src/main/kotlin/com/mthaler/bankapp/dao/CustomerRequestDaoImpl.java
package com.mthaler.bankapp.dao;

import com.mthaler.bankapp.domain.CustomerRequestDetails;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class CustomerRequestDaoImpl implements CustomerRequestDao {
	private static Logger logger = LogManager.getLogger(CustomerRequestDaoImpl.class);
	
	public CustomerRequestDaoImpl() {
		logger.info("Created CustomerRequestDaoImpl instance");
	}
	
	@Override
	public void submitRequest(CustomerRequestDetails userRequestDetails) {
		// --save UserRequestDetails
	}

}

File: annotations/src/main/kotlin/com/mthaler/bankapp/dao/CustomerRequestDaoImpl.kt
package com.mthaler.bankapp.dao

import com.mthaler.bankapp.dao.CustomerRequestDao
import com.mthaler.bankapp.domain.CustomerRequestDetails
import com.mthaler.bankapp.dao.CustomerRequestDaoImpl
import org.apache.logging.log4j.LogManager
import org.springframework.stereotype.Repository

@Repository(value = "customerRequestDao")
class CustomerRequestDaoImpl : CustomerRequestDao {

    init {
        logger.info("Created CustomerRequestDaoImpl instance")
    }

    override fun submitRequest(userRequestDetails: CustomerRequestDetails) {
        // --save UserRequestDetails
    }

    companion object {
        private val logger = LogManager
            .getLogger(CustomerRequestDaoImpl::class.java)
    }
}